libraries/AP_HAL_Linux/SPIDevice.cpp

```diff
@@ -164,21 +169,26 @@ class SPIBus : public TimerPollable::WrapperCb {
     void start_cb() override;
     void end_cb() override;
 
-    int open(uint16_t bus_, uint16_t kernel_cs_);
+    void open(uint16_t subdev);
 
     PollerThread thread;
     Semaphore sem;
-    int fd = -1;
+    int fd[MAX_SUBDEVS];
     uint16_t bus;
-    uint16_t kernel_cs;
-    uint8_t ref;
     int16_t last_mode = -1;
+    uint8_t ref;
 };
 
+SPIBus::SPIBus(uint16_t bus_)
+    : bus(bus_)
+{
+    memset(fd, -1, sizeof(fd));
```

Comment:
is using "-1" normal here for Linux? I would think either 0xFF or loop and set to -1

Reply:
Memset works with ints